### ISSUE TITLE
 `in_batch_shuffle` should respect shuffle settings (on/off and determinism)

### ISSUE THREAD

**Issue by VitalyFedyunin** [pytorch/data#739]
> Looks like `in_batch_shuffle` would be a good alternative, however I do need the ability to disable shuffling because the same code-path would be used for the training sets (shuffle enabled) and the tests sets (shuffle disabled).

Yeah. This is one of TODOs on top of my lists. I need to align `in_batch_shuffle` with `shuffle` in terms of API.

Edit: `shuffle_setting` should also respect those two types of `DataPipe`.

_Originally posted by @ejguan in https://github.com/pytorch/data/issues/732#issuecomment-1216608149_

**Comment by ejguan** [comment]
Closing this issue as both https://github.com/pytorch/pytorch/pull/83741 and https://github.com/pytorch/data/pull/745 are landed.
